### ISSUE TITLE
cmd/compile,runtime: frequent test timeouts on solaris-amd64-oraclerel

### ISSUE THREAD

**Issue by bcmills** [golang/go#51443]
```
#!watchflakes
default <- builder == "solaris-amd64-oraclerel" && (pkg == "runtime" || pkg == "cmd/compile") && (`^panic: test timed out` || `failed to start: context deadline exceeded`)
```

`greplogs --dashboard -md -l -e '(?ms)\Asolaris-amd64-oraclerel.*^panic: test timed out.*FAIL\s+runtime' --since=2022-01-01`

[2022-03-01T21:27:42-b0db2f0/solaris-amd64-oraclerel](https://build.golang.org/log/58f45d6253afba9ff06a879dc545bf30ce07b71f)
[2022-02-28T19:00:23-b33592d/solaris-amd64-oraclerel](https://build.golang.org/log/4ffce9f38e803689007102af60fd26be84a00f77)
[2022-01-08T00:24:25-90860e0/solaris-amd64-oraclerel](https://build.golang.org/log/3dce7054be085f7d06de90a6e07452e31546d022)
[2022-01-06T23:39:43-042548b/solaris-amd64-oraclerel](https://build.golang.org/log/474d50f9d379f7a607268933e5e5f4c5c99cbb4f)

@golang/runtime: the builder appears to only run the `-short` tests. Is there something we can feasibly do to make `-short` mode shorter?

@golang/release, @rorth: would it make sense to set `GO_TEST_TIMEOUT_SCALE` on this builder to make the timeouts more generous?

**Comment by dmitshur** [comment]
It makes sense to increase `GO_TEST_TIMEOUT_SCALE` to me, but @rorth as the [builder owner](https://farmer.golang.org/builders) is in a better position to decide the optimal configuration for this builder. This builder is not configured as a TryBot, so it if it needs more time, it's fine and won't affect pre-submit testing times (#17104).

**Comment by rorth** [comment]
"Bryan C. Mills" ***@***.***> writes:

> @golang/release, @rorth: would it make sense to set `GO_TEST_TIMEOUT_SCALE`
> on this builder to make the timeouts more generous?

I'd be ok with that, but wondered about one thing: that zone is shared
between the golang builder and an LLVM buildbot.  The latter is
restricted to 8 parallel jobs, and I'd been under the impression that
the golang one would work with just 4 jobs.  However, recently I've
often seen way more than those 4 jobs.  Have I just been deluded about
that restriction, or has it somehow been lost?

**Comment by ianlancetaylor** [comment]
While running tests, a go builder will happily grab whatever resources it has available.  Are you doing anything to restrict it to 4 jobs?

**Comment by rorth** [comment]
Hi Ian,

> While running tests, a go builder will happily grab whatever resources it
> has available.  Are you doing anything to restrict it to 4 jobs?

Not so far.  When I inherited the builder, I was under the (obviously
wrong) impression that this had been done in the builder config,
similarly to buildbot for LLVM.  So there's no similar facility
available?

I should be able to use CPU shares to achieve this.  Setting up a
separate zone just for the golang builder seems like a waste of effort
(and IPv4 addresses).

**Comment by ianlancetaylor** [comment]
Good point, I think the builder config will set `GOMAXPROCS`, which will limit the resource usage of each specific test, but then you will have 4 tests run in parallel and each test will happily run 4 tests in parallel.  At least, I think that is how it works; my apologies if I'm getting this wrong.

**Comment by rorth** [comment]
Ian Lance Taylor ***@***.***> writes:

> Good point, I think the builder config will set `GOMAXPROCS`, which will
> limit the resource usage of each specific test, but then you will have 4
> tests run in parallel and each test will happily run 4 tests in parallel.
> At least, I think that is how it works; my apologies if I'm getting this
> wrong.

As a first step, I've now set GOMAXPROCS=8 for the builder to see if
this helps.  The last runtime timeout happened before that change,
though.  We'll have to see if this is enough.

In parallel, I'm also running the runtime test 10000 times in a loop on
identical hardware (although on bare metal instead of a kernel
zone/VM).  Of the roughly 4000 runs completed so far, 6 have taken 2+
minutes rather than the usual 16-18 seconds.  I missed the
-test.timeout=3m, unfortunately, so all tests PASSed.

Unfortunately, I couldn't easily see which subtest is timing out,
otherwise I could restrict the testing to just that one if it's always
the same one.

**Comment by bcmills** [comment]
> As a first step, I've now set GOMAXPROCS=8 for the builder to see if
> this helps.  The last runtime timeout happened before that change,
> though.  We'll have to see if this is enough.

Appears not to be:

`greplogs --dashboard -md -l -e '(?ms)\Asolaris-amd64-oraclerel.*^panic: test timed out.*FAIL\s+runtime' --since=2022-03-08`

[2022-03-29T16:24:51-a2baae6/solaris-amd64-oraclerel](https://build.golang.org/log/3dd30b3f4146a21ee04cf7adc4e16d85a67667cc)
[2022-03-25T19:04:59-2bbf383/solaris-amd64-oraclerel](https://build.golang.org/log/7abedc726c419ecd05628fd72b15fe2457e902fb)
[2022-03-25T15:15:57-3fd8b86/solaris-amd64-oraclerel](https://build.golang.org/log/51b19d8fdcbad15ec1c35b3df91c310d2d790631)

**Comment by bcmills** [comment]
`greplogs --dashboard -md -l -e '(?ms)\Asolaris-amd64-oraclerel.*^panic: test timed out.*FAIL\s+runtime' --since=2022-03-30`
[2022-05-02T19:53:56-3b01a80/solaris-amd64-oraclerel](https://build.golang.org/log/e2686bc7eb06a28e7848ca348665c73fce4f12c1)

**Comment by bcmills** [comment]
`greplogs -l -e '(?ms)\Asolaris-amd64-oraclerel.*^panic: test timed out.*FAIL\s+runtime' --since=2022-05-03`
[2022-05-11T03:28:01-ccb7987/solaris-amd64-oraclerel](https://build.golang.org/log/bf9ffb5c1d70cd184c51747b2dd221f116eea1ad)
[2022-05-10T21:56:21-bda9da8/solaris-amd64-oraclerel](https://build.golang.org/log/7aad7d9b7c815473c777bbf04ad60dfc55df2f18)

**Comment by bcmills** [comment]
`greplogs -l -e '(?ms)\Asolaris-amd64-oraclerel.*^panic: test timed out.*FAIL\s+runtime' --since=2022-05-12`
[2022-05-25T19:25:08-04337a6/solaris-amd64-oraclerel](https://build.golang.org/log/99321323f0d0eed0f3c18a1a14ceea85923f126c)
[2022-05-23T16:21:22-c1d197a/solaris-amd64-oraclerel](https://build.golang.org/log/c2fbdfe3860e337fc24f0b338109a6afbf12e623)

**Comment by bcmills** [comment]
There have apparently not been any more of these failures since May. I'm calling this obsolete.

**Comment by mknyszek** [comment]
In triage, it looks to us like all the tests are just timing out. The builder just seems slow. There might be one culprit. Verbose test output would tell us which test cases are slow.

CC @golang/solaris

**Comment by bcmills** [comment]
The C toolchain on this builder seems particularly slow. I wonder if just raising the `GO_TEST_TIMEOUT_SCALE` on that builder might resolve things.

**Comment by rorth** [comment]
"Bryan C. Mills" ***@***.***> writes:

> The C toolchain on this builder seems particularly slow. I wonder if just
> raising the `GO_TEST_TIMEOUT_SCALE` for the builder might resolve things.

The buildlet is running in a kernel zone (VM), and for some reason some
of the tests run extremely slow there, compared to bare metal on an
identical server or the host system.

For the time being, I've increased GO_TEST_TIMEOUT_SCALE to 4 as a
workaround.

Besides, I hope to migrate the buildlet to a different system (5 year
old 2 x Xeon Gold 5120 instead of 12 year old 4 x Xeon E7540) in the
coming months.

**Comment by bcmills** [comment]
Duplicate of #60152
